### PR TITLE
Additional improvements to call_slack_api()

### PR DESCRIPTION
**Pull request by andrie**
This includes pagination, improved error handling and fixes Auth for GET as well as POST requests.

### CODE REVIEW

File: R/slackr_setup.r
Comment (mrkaye97) [inline R/slackr_setup.r:106]: Do we want to return `NULL` here or some kind of success message?

File: R/slackr_utils.R
Comment (mrkaye97) [inline R/slackr_utils.R:93]: Nice. I like this change.

File: R/slackr_utils.R
Comment (mrkaye97) [inline R/slackr_utils.R:105]: This is great too. Simplifies things a lot.

File: R/slackr_utils.R
Comment (mrkaye97) [inline R/slackr_utils.R:163]: :+1: Having this split up into private + public seems like a good idea to me

File: R/call_slack_api.R
Comment (mrkaye97) [inline R/call_slack_api.R:109]: what does the `isTRUE()` add here? seems like just `is.na()` should be sufficient?

File: R/call_slack_internals.R
Comment (mrkaye97) [inline R/call_slack_internals.R:43]: is this supposed to supercede `slackr_msg()` / `text_slackr()`?

File: R/call_slack_internals.R
Comment (mrkaye97) [inline R/call_slack_internals.R]: Is this a duplicate? I can't tell if there's some diff

File: R/call_slack_internals.R
Comment (mrkaye97) [inline R/call_slack_internals.R:101]: Will this one supercede `slackr_upload()` and `save_slackr()`?

File: DESCRIPTION
Comment (mrkaye97) [inline DESCRIPTION:38]: I think we'll probably need to add a `magrittr` dep if we're using the `%>%` in a bunch of places. I'm in favor of that. I think that being more tidy-forward is a good idea. Also will need `tibble` I believe.

I'm surprised the checks are passing without those being included here. Is there a reason for that?

**Review by mrkaye97 — APPROVED** [review]
This looks awesome! I left some comments in different spots, but I think it's great overall. I really like the idea to factor out the API calls. That should make the whole package be a lot easier to iterate on, test, etc. I also think being more tidy-friendly is a good strategy, so I'm happy we're making more use of `tibble`, `dplyr`, `magrittr`, etc. 

I have a PR open with a bunch of changes too, but I think a lot of them will conflict with these. I'll hold off on merging those until you merge so I can fix the conflicts, unless you'd prefer to do it (lol). That PR adds some tests and a couple new functions (`slackr_delete()` and `slackr_history()`), but the tests won't pass in CI until Bob transfers the repo over to me so that I can add secrets.

Thanks again for all the work on this! It looks great

File: R/call_slack_api.R
Comment (andrie) [reply R/call_slack_api.R:109]: Use `isTRUE()` to ensure the resulting vector has length 1.  If you relied on `is.na()` and `result` is a vector, then `is.na()` will operate on every element of that vector.  For an `if()` condition you want to be sure the element has length 1.

File: R/call_slack_internals.R
Comment (andrie) [reply R/call_slack_internals.R:43]: No, not directly.  These functions are all internal.  Thus they have minimal checking for valid inputs and handling exceptions.  The existing exported functions should then make use of these internal functions.

File: R/call_slack_internals.R
Comment (andrie) [reply R/call_slack_internals.R]: Good catch.  This was an errant copy/paste.  I've removed this.

File: R/call_slack_internals.R
Comment (andrie) [reply R/call_slack_internals.R:101]: No. This is an internal function that will be used by many other functions. At the moment I've only tested this with `ggslackr` but it should also work with `dev_slackr()` and eventually these functions you mention.

At the moment I'm still trying to not change the exported API.

File: DESCRIPTION
Comment (andrie) [reply DESCRIPTION:38]: Agreed.  I'll  add the pipe and dependency on `tibble` this morning.

The reason that `R CMD CHECK` doesn't complain is because right now the package imports all of this:

```
import(dplyr)
import(ggplot2)
import(httr)
import(methods)
import(utils)
```

Since all of `dplyr` is imported, the pipe is also available.

Importing entire packages, rather than just selected functions, is a bad idea.  We need to fix this.

**Pull request merged**
